lib/matplotlib/type1font.py

```diff
@@ -125,20 +130,76 @@ def _split(self, data):
                 zeros -= 1
             idx -= 1
         if zeros:
-            raise RuntimeError('Insufficiently many zeros in Type 1 font')
+            # this may have been a problem on old implementations that
+            # used the zeros as necessary padding
+            _log.info('Insufficiently many zeros in Type 1 font')
 
         # Convert encrypted part to binary (if we read a pfb file, we may end
         # up converting binary to hexadecimal to binary again; but if we read
         # a pfa file, this part is already in hex, and I am not quite sure if
         # even the pfb format guarantees that it will be in binary).
-        binary = binascii.unhexlify(data[len1:idx+1])
+        idx1 = len1 + ((idx - len1 + 2) & ~1)  # ensure an even number of bytes
+        binary = binascii.unhexlify(data[len1:idx1])
 
         return data[:len1], binary, data[idx+1:]
 
     _whitespace_or_comment_re = re.compile(br'[\0\t\r\014\n ]+|%[^\r\n\v]*')
     _token_re = re.compile(br'/{0,2}[^]\0\t\r\v\n ()<>{}/%[]+')
     _instring_re = re.compile(br'[()\\]')
 
+    @staticmethod
+    def _decrypt(ciphertext, key, ndiscard=4):
+        """
+        Decrypt ciphertext using the Type-1 font algorithm
+
+        The algorithm is described in Adobe's "Adobe Type 1 Font Format".
+        The key argument can be an integer, or one of the strings
+        'eexec' and 'charstring', which map to the key specified for the
+        corresponding part of Type-1 fonts.
+
+        The ndiscard argument should be an integer, usually 4.
+        That number of bytes is discarded from the beginning of plaintext.
+        """
+
+        key = {'eexec': 55665, 'charstring': 4330}.get(key, key)
+        if not isinstance(key, int):
+            raise ValueError(f'Invalid decryption key {key!r}')
+
+        plaintext = bytearray(len(ciphertext))
```

Comment:
I would have done a boring `plaintext = []` and later `plaintext.append(...)` skipping the need for enumerate and finally `return b"".join(...)`?